sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java

```diff
@@ -372,6 +372,113 @@ public void replaceEndToEnd() throws JSONException {
         assertEquals(TEST_FIELD_VALUE, testDocument.test);
     }
 
+    @Test
+    public void readFailsToDeserializeDocumentDoesNotThrow() throws JSONException {
+        /* Mock http call to get token. */
+        /* Pass incorrect document type to cause serialization failure (document is of type TestDocument). */
+        AppCenterFuture<Document<String>> doc = Storage.read(Constants.USER, DOCUMENT_ID, String.class);
+
+        /* Make cosmos db http call with exchanged token. */
+        verifyTokenExchangeFlow(TOKEN_EXCHANGE_USER_PAYLOAD, null);
+        verifyCosmosDbFlow(DOCUMENT_ID, METHOD_GET, COSMOS_DB_DOCUMENT_RESPONSE_PAYLOAD, null);
+
+        /* Get and verify document. Confirm the cache was not touched. */
+        assertNotNull(doc);
+        verifyNoMoreInteractions(mLocalDocumentStorage);
+        Document<String> testCosmosDocument = doc.get();
+        assertNotNull(testCosmosDocument);
+        assertTrue(testCosmosDocument.failed());
+    }
+
+    @Test
+    public void createFailsToDeserializeDocumentDoesNotThrow() throws JSONException {
```

Comment:
Thats the JSONException from built in json library, not Gson, why is it here?

Reply:
It's because`verifyTokenExchangeToCosmosDbFlow` mocks the http calls in a way that can throw (down on lines 278 and 307)

Reply:
okay